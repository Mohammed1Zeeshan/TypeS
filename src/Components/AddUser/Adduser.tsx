import axios from 'axios'
import React, { useState } from 'react'
import { Button, Form } from 'react-bootstrap'
import { useHistory } from 'react-router-dom'
import { Input } from 'reactstrap'


function Adduser() {

  const historyy=useHistory()
    
    const [users,setUsers]=useState({
       // id:'',
        key:'',
        value:''
    })
    const {key,value}=users;

    
    const onInputChange =(e: React.ChangeEvent<HTMLInputElement>)=>{setUsers({...users,[e.target.name]: e.target.value});console.log(users);}

    //const Submit= async (e: { preventDefault: () => void })=>{e.preventDefault();await axios.post('http://localhost:3005/users',user);historyy.push('')}
    const Submit= async (e: { preventDefault: () => void })=>{e.preventDefault();await axios.post('https://localhost:44330/api/User/Addstu',users);historyy.push('')}


  return (
    <div className='constainer'><h1>Add User</h1>
      <div className='w-75 mx-auto shadow p-5'>
        <Form>
          <div className='form-group'>
          <Input type='text'  placeholder="enter Key"  name="key" className='form-control form-control-lg'
           value={key} onChange={e=>onInputChange(e)} ></Input></div> 
        </Form>
        <Form>
          <div className='form-group'>
          <Input type='text'  placeholder="enter Value"  name="value" className='form-control form-control-lg'
            value={value} onChange={e=>onInputChange(e)}></Input></div> 
        </Form>
        <Button onClick={Submit} >Add</Button>
      </div>
    </div>
  )
}

export default Adduser
