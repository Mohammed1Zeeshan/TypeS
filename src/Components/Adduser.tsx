import axios from 'axios'
import React, { useState } from 'react'
import { Button, Form } from 'react-bootstrap'
import { useHistory } from 'react-router-dom'
import { Input } from 'reactstrap'


function Adduser() {

  const historyy=useHistory()
    
    const [user,setUser]=useState({
        id:'',
        Key:'',
        Value:''
    })
    const {id,Key,Value}=user;

    const onInputChange =(e: React.ChangeEvent<HTMLInputElement>)=>{setUser({...user,[e.target.name]: e.target.value});console.log(user);}

    const Submit= async (e: { preventDefault: () => void })=>{e.preventDefault();await axios.post('http://localhost:3005/users',user);historyy.push('')}

  return (
    <div className='constainer'><h1>Add User</h1>
      <div className='w-75 mx-auto shadow p-5'>
        <Form>
          <div className='form-group'>
          <Input type='text'  placeholder="enter Key"  name="Key" className='form-control form-control-lg'
           value={Key} onChange={e=>onInputChange(e)} ></Input></div> 
        </Form>
        <Form>
          <div className='form-group'>
          <Input type='text'  placeholder="enter Value"  name="Value" className='form-control form-control-lg'
            value={Value} onChange={e=>onInputChange(e)}></Input></div> 
        </Form>
        <Button onClick={Submit} >Add</Button>
      </div>
    </div>
  )
}

export default Adduser
