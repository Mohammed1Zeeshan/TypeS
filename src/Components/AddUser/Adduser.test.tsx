import {shallow } from "enzyme";
import Adduser from "./Adduser";
import puppeteer from 'puppeteer-core';
import {toMatchImageSnapshot} from 'jest-image-snapshot';

jest.setTimeout(30000)

expect.extend({toMatchImageSnapshot});
const browserPath="C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe";


describe('shallow', ()=>{
    it('shallow should render the component correctly',()=>{
        const wraper=shallow(<Adduser/>)
        expect(wraper).toMatchSnapshot();
    })
})
// it('Google page',async()=>{
  //   const browser = await puppeteer.launch({executablePath:browserPath,});
  //   const page=await browser.newPage();
  //   await page.goto('https://www.google.com/');
  //   const image= await page.screenshot()
  //   expect(image).toMatchImageSnapshot();
  // });


    it('create add user snapshot',async()=>{
    const browser = await puppeteer.launch({executablePath:browserPath,});
    const page=await browser.newPage();
    await page.goto('http://localhost:3006/Adduser');
    const image= await page.screenshot()
    expect(image).toMatchImageSnapshot();
  });
