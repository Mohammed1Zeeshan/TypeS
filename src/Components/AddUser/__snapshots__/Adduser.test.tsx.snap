// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow shallow should render the component correctly 1`] = `
<div
  className="constainer"
>
  <h1>
    Add User
  </h1>
  <div
    className="w-75 mx-auto shadow p-5"
  >
    <Form>
      <div
        className="form-group"
      >
        <Input
          className="form-control form-control-lg"
          name="Key"
          onChange={[Function]}
          placeholder="enter Key"
          type="text"
          value=""
        />
      </div>
    </Form>
    <Form>
      <div
        className="form-group"
      >
        <Input
          className="form-control form-control-lg"
          name="Value"
          onChange={[Function]}
          placeholder="enter Value"
          type="text"
          value=""
        />
      </div>
    </Form>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Add
    </Button>
  </div>
</div>
`;
